### PR TITLE
Move remaining InteractiveWindow behavior to the UI thread

### LINKED ISSUE #4044: Review all mutable fields of InteractiveWindow
Unlike the fields of ```InteractiveWindow.UIThreadOnly```, the fields of ```InteractiveWindow``` can be (and are) accessed by any thread.  Since we're not doing anything to sequentialize accesses, we probably have race conditions.

### PR DESCRIPTION
**Pull request by amcasey**
Fixes #4044.

### CODE REVIEW

**Comment by amcasey** [comment]
FYI @cston @KevinH-MS @ManishJayaswal @tmat @DinoV

File: src/InteractiveWindow/Editor/InteractiveWindow.cs [hunk 1]
```diff
@@ -32,114 +33,107 @@ namespace Microsoft.VisualStudio.InteractiveWindow
...
 
-        ////
-        //// Services
-        //// 
+        PropertyCollection IPropertyOwner.Properties { get; } = new PropertyCollection();
 
-        private readonly IIntellisenseSessionStackMapService _intellisenseSessionStackMap;
-        private readonly ISmartIndentationService _smartIndenterService;
+        private readonly SemaphoreSlim _inputReaderSemaphore = new SemaphoreSlim(initialCount: 1, maxCount: 1);
+        private readonly UIThreadOnly _dangerous_uiOnly;
```
Comment (cston) [inline src/InteractiveWindow/Editor/InteractiveWindow.cs]: Perhaps '_uiOnly'.

File: src/InteractiveWindow/Editor/InteractiveWindow.cs [hunk 2]
```diff
@@ -513,56 +401,9 @@ private static void DumpCmd(string prefix, int result, ref Guid pguidCmdGroup, u
 
         #region Caret and Cursor
 
-        private ITextCaret Caret => _textView.Caret;
-
         private void CaretPositionChanged(object sender, CaretPositionChangedEventArgs e)
```
Comment (cston) [inline src/InteractiveWindow/Editor/InteractiveWindow.cs]: Is this event handler needed on the outer (non-UI thread) object?

Comment (cston) [inline src/InteractiveWindow/Editor/InteractiveWindow.cs]: Is this event handler needed on the outer (non-UI thread) object? Events should only occur on the UI thread.

**Comment by cston** [comment]
LGTM

File: src/InteractiveWindow/Editor/InteractiveWindow.cs (hunk 1 above)
Comment (amcasey) [inline src/InteractiveWindow/Editor/InteractiveWindow.cs]: Why?  You still have to think carefully every time you use it.

File: src/InteractiveWindow/Editor/InteractiveWindow.cs (hunk 2 above)
Comment (amcasey) [inline src/InteractiveWindow/Editor/InteractiveWindow.cs]: There doesn't seem to be a guarantee that the event will be raised on the UI thread and marshalling is the outer object's responsibility.

**Comment by amcasey** [comment]
New push was just to resolve merge conflicts.

File: src/InteractiveWindow/Editor/InteractiveWindow.cs (hunk 1 above)
Comment (cston) [inline src/InteractiveWindow/Editor/InteractiveWindow.cs]: The 'uiOnly' part captures that.

Comment (amcasey) [inline src/InteractiveWindow/Editor/InteractiveWindow.cs]: Done.

File: src/InteractiveWindow/Editor/InteractiveWindow.cs (hunk 2 above)
Comment (amcasey) [inline src/InteractiveWindow/Editor/InteractiveWindow.cs]: Per our offline discussion, this is a generally-accepted assumption.

**Pull request merged**